### DISCUSSION TITLE
How do I skip missing properties in request input while parsing

### DISCUSSION THREAD

**Discussion by pulkitvyas08** [colinhacks/zod#1684] [Q&A] [answered]
I used class-validator before and there we can pass a `skipMissingProperties` flag to only validate the properties that are present in the input request body and ignore if the rest are missing. Is the equivalent of that present in Zod? If not then any ideas how can I implement this? `safeParse` doesn't seem to work here

**Comment by JacobWeisenburger** [comment, marked as answer]
Here are a few parts of the docs that might help:

- [`.optional`](https://github.com/colinhacks/zod#optionals)
- [`.partial`](https://github.com/colinhacks/zod#partial)
- [`.deepPartial`](https://github.com/colinhacks/zod#deeppartial)

Let me know if you have any questions about these.

**Reply by pulkitvyas08** [reply]
Yes this work, thank you!
